Prog/Affordability_sf_condo.sas

```diff
@@ -0,0 +1,321 @@
+	if saleyear =2001 then do;
+           if PITI_First <= (85600*0.8 / 12*.28) then AMI80_first_afford=1; else AMI80_first_afford=0;
+		   if PITI_First <= (85600*0.5 / 12*.28) then AMI50_first_afford=1; else AMI50_first_afford=0;
+		   if PITI_Repeat <= (85600*0.8 / 12*.28) then AMI80_repeat_afford=1; else AMI80_repeat_afford=0;
+		   if PITI_Repeat <= (85600*0.5 / 12*.28) then AMI50_repeat_afford=1; else AMI50_repeat_afford=0;
+	end;
+	if saleyear =2000 then do;
+           if PITI_First <= (82800*0.8 / 12*.28) then AMI80_first_afford=1; else AMI80_first_afford=0;
+		   if PITI_First <= (82800*0.5 / 12*.28) then AMI50_first_afford=1; else AMI50_first_afford=0;
+		   if PITI_Repeat <= (82800*0.8 / 12*.28) then AMI80_repeat_afford=1; else AMI80_repeat_afford=0;
+		   if PITI_Repeat <= (82800*0.5 / 12*.28) then AMI50_repeat_afford=1; else AMI50_repeat_afford=0;
+	end;
+
+end;
+
+run;
+proc contents data=create_flags;
+run;
+
+proc print data= create_flags (obs=25);
+var saleprice saleyear PITI_FIRST PITI_repeat AMI80_first_afford AMI50_first_afford AMI80_repeat_afford AMI50_repeat_afford;
+run;
+proc freq data=create_flags; 
+tables AMI80_first_afford AMI50_first_afford AMI80_repeat_afford AMI50_repeat_afford; 
+run;
+*proc summary at city, ward levels - so you could get % of sales in Ward 7 affordable to 
+median white family vs. median black family.;
+
+/*Proc Summary: Affordability for Owners by 80AMI and 50AMI*/
+
+proc summary data=create_flags;
+	class city saleyear;
+	var total_sales AMI80_first_afford AMI50_first_afford AMI80_repeat_afford AMI50_repeat_afford;
+	output	out=City_level	sum= ;
+	format city $CITY16.;
+		run;
+
+proc summary data=create_flags;
+	class ward2012 saleyear;
+	var total_sales AMI80_first_afford AMI50_first_afford AMI80_repeat_afford AMI50_repeat_afford;
+	output 	out=Ward_Level 
+	sum= ; 
+	format ward2012 $wd12.;
+;
+		run;
+
+	data sales_afford_SF_Condo (label="DC Single Family Home Sales Affordabilty for 80%, 50% Area Median Income, 2000-17" drop=_type_ _freq_);
+
+	set city_level ward_level; 
+
+	PctAffordFirst_80AMI=AMI80_first_afford/total_sales*100; 
+	PctAffordFirst_50AMI=AMI50_first_afford/total_sales*100; 
+    PctAffordRepeat_80AMI=AMI80_repeat_afford/total_sales*100; 
+	PctAffordRepeat_50AMI=AMI50_repeat_afford/total_sales*100; 
+
+	/*add code for repeat buyer affordability*
+	PctAffordRepeat_80AMI=80AMI_repeat_afford/total_sales*100; 
+*/
+	label PctAffordFirst_80AMI="Pct. of SF/Condo Sales 2000-17 Affordable to 80% AMI and 50% AMI"
+			;
+	run;
+
+	
+proc export data=sales_afford_SF_Condo
+	outfile="D:\DCDATA\Libraries\DMPED\Prog\sf_condo_tabs_aff.csv"
+	dbms=csv replace;
+	run;
+
+	
+	** Register metadata **;
+%Dc_update_meta_file(
+      ds_lib=DMPED,
+      ds_name=sales_afford_SF_Condo,
+      creator_process=Affordability_sf_condo.sas,
+      restrictions=None,
+      revisions=New file.
+      )
+
+data wardonly;
+	set DMPED.sales_afford_SF_Condo (where=(ward2012~=" ") keep=ward2012 pct:); 
```

Comment:
@maggiesu0725 - just keep it a temporary file

Reply:
@maggiesu0725 - you can transpose the city and ward together - should be easier once you replace the class statements with the by statements